main.js

```diff
@@ -1,18 +1,40 @@
 import messenger from './src/js/postMessenger';
 
+const sendMonitoringEvent = message =>
+	messenger.post({ type: 'oAds.monitor', message }, window.top);
+
 const handleReceivedMessage = event => {
-	if (event.origin === 'https://ft.com' && event.data) {
-		if (event.data.messageType === 'oAdsEmbed') {
+	if (event.data && event.data.messageType === 'oAdsEmbed') {
+		if (event.origin === 'https://ft.com') {
 			window.oAdsEmbedData = event.data.body;
+		} else {
+			sendMonitoringEvent(`oAdsEmbed message from unexpected origin: ${event.origin}`);
 		}
 	}
 };
 
+const checkSmartmatchProp = () => {
+	// Is this code running on a Smartmatch-compatible page?
+	const pageUrl = window.top.location && window.top.location.href;
+	if (!pageUrl) {
+		sendMonitoringEvent('Top window location info inaccessible');
+		return;
+	}
+
+	const isSMpage = pageUrl.match(/ft.com\/content/);
+
+	if (isSMpage) {
+		const hasSMObjOnLoad = Boolean(window.top.smartmatchCreativeMatches);
+		const neg = hasSMObjOnLoad ? ' ' : ' NOT ';
+		sendMonitoringEvent(`SM obj was${neg}available when iframe loaded`);
```

Comment:
Do it need to be monitored also when SM object is successfully retrieved?

Reply:
Yes. I need to compare total vs failures